tests/main/test_image_model_wrapper.py

```diff
@@ -137,12 +140,64 @@ def test_pytorch_object_detection_model_pandas(self):
     @pytest.mark.skipif(sys.version_info.minor <= 6,
                         reason='Older versions of pytorch not supported')
     def test_pytorch_object_detection_custom_model_pandas(self):
+        model, data = self._set_up_OD_model()
+        wrapped_model = PytorchDRiseWrapper(model, 1)
+        validate_wrapped_object_detection_custom_model(wrapped_model,
+                                                       T.ToTensor()(data[0])
+                                                       .repeat(2, 1, 1, 1))
+
+    # Skip for older versions of pytorch due to missing classes
+    @pytest.mark.skipif(sys.version_info.minor <= 6,
+                        reason='Older versions of pytorch not supported')
+    def test_PytorchDRiseWrapper_wrapper_device(self):
+        model, data = self._set_up_OD_model()
+
+        wrapped_model = PytorchDRiseWrapper(model, 1, 'cpu')
+        validate_wrapped_object_detection_custom_model(wrapped_model,
+                                                       T.ToTensor()(data[0])
+                                                       .repeat(2, 1, 1, 1))
+        if torch.cuda.is_available():
+            wrapped_model = PytorchDRiseWrapper(model, 1, 'cuda')
+            validate_wrapped_object_detection_custom_model(
+                wrapped_model,
+                T.ToTensor()(data[0])
+                .repeat(2, 1, 1, 1))
+        else:
+            with pytest.raises(ValueError,
+                               match="Selected device is invalid"):
+                wrapped_model = PytorchDRiseWrapper(model, 1, 'cuda')
+
+    # Skip for older versions of pytorch due to missing classes
+    @pytest.mark.skipif(sys.version_info.minor <= 6,
+                        reason='Older versions of pytorch not supported')
+    def test_WrappedObjectDetectionModel_wrapper_device(self):
+        model, data = self._set_up_OD_model()
+        wrapped_model = WrappedObjectDetectionModel(model, 1, 'cpu')
+        validate_wrapped_object_detection_custom_model(wrapped_model,
+                                                       T.ToTensor()(data[0])
+                                                       .repeat(2, 1, 1, 1))
+        if torch.cuda.is_available():
+            wrapped_model = WrappedObjectDetectionModel(model, 1, 'cuda')
+            validate_wrapped_object_detection_custom_model(
+                wrapped_model,
+                T.ToTensor()(data[0])
+                .repeat(2, 1, 1, 1))
+        else:
+            with pytest.raises(ValueError,
+                               match="Selected device is invalid"):
+                wrapped_model = WrappedObjectDetectionModel(model, 1, 'cuda')
+
+    def test_get_device(self):
+        # test default invocation of _get_device as it would be during the
+        # wrap_model invocation in RAIVisionInsights
+        device = _get_device("auto")
+        assert device == "cpu"
```

Comment:
this may not pass on all machines for all users running the tests in this repo - maybe we can just assert if it's cpu or gpu?